provisioner/terraform/install_test.go

```diff
@@ -101,8 +145,82 @@ func mainJSON(versions ...*version.Version) product {
 	return mj
 }
 
-func exeContent(v *version.Version) []byte {
-	return []byte(strings.ReplaceAll(terraformExecutableTemplate, "${ver}", v.String()))
+// for linux/mac simple script works
+func unixExeContent(t *testing.T, v *version.Version, platform osArch) []byte {
+	rep := strings.NewReplacer("${ver}", v.String(), "${os}", platform.os, "${arch}", platform.arch)
+	return []byte(rep.Replace(bashExecutableTemplate))
+}
+
+// for windows it seems some progmram complication is required
+func windowsExeContent(t *testing.T, tmpDir string, v *version.Version, platform osArch) []byte {
+	code := fmt.Sprintf(windowsExecutableGoSourceCodeTemplate, v, platform.arch)
+	require.NoError(t, os.WriteFile(filepath.Join(tmpDir, "fake-terraform.go"), []byte(code), 0o600))
+
+	var errbuf strings.Builder
+	if _, err := os.Stat(filepath.Join(tmpDir, "go.mod")); errors.Is(err, os.ErrNotExist) {
+		cmd := exec.Command("go", "mod", "init", "fake-terraform")
```

Comment:
Thinking loud -

Since you have logic to dynamically create a fake terraform binary, can we use it for all cases, I mean mac/linux/windows? 
Also, if the binary runs on the same platform as testing code, can it dynamically figure out the os and arch?

I'm curious if we can get rid of those if-windows clauses in favor of generic code for all platforms.

Reply:
Talked with @hugodutka a bit and decided to go with cached proxy approach. Installers are cached in test cache and only downloaded on cache miss.